tests/reftests/simulated-pin.test

```diff
@@ -0,0 +1,274 @@
+N0REP0
+### OPAMYES=1
+### ::: Tests specific to simulated pinning behaviour, trigerred by `opam install ./localdir --deps-only`
+### :I: Checking the opam file update reinstall trigger
+### <pkg:reinst.1>
+opam-version: "2.0"
+depends: "dep-reinst" { = "1" }
+### <pkg:dep-reinst.1>
+opam-version: "2.0"
+### <pkg:dep-reinst.2>
+opam-version: "2.0"
+### <pkg:dep-reinst.3>
+opam-version: "2.0"
+### :I:1: Update opam file and install again
+### opam switch create reinst-1 --empty
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "1" }
+### opam install ./reinst
+reinst is now pinned to file://${BASEDIR}/reinst (version 1)
+The following actions will be performed:
+=== install 2 packages
+  - install dep-reinst 1          [required by reinst]
+  - install reinst     1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed dep-reinst.1
+-> retrieved reinst.1  (file://${BASEDIR}/reinst)
+-> installed reinst.1
+Done.
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "2" }
+### opam install ./reinst --deps-only
+[reinst.1] synchronised (file://${BASEDIR}/reinst)
+[reinst] Installing new package description from upstream file://${BASEDIR}/reinst
+The following actions will be performed:
+=== recompile 1 package
+  - recompile reinst     1 (pinned) [uses dep-reinst]
+=== upgrade 1 package
+  - upgrade   dep-reinst 1 to 2     [required by reinst]
```

Comment:
cf https://github.com/ocaml/opam/pull/6522#discussion_r2108984515
>  This should be rejected because the dependency constraints from the pinned package and the requested deps are conflicting.

Issue #6529 & fixed #6530